rusqlite_migration/src/loader.rs

```diff
@@ -95,50 +91,47 @@ impl<'u> From<&MigrationFile> for M<'u> {
 }
 
 pub(crate) fn from_directory(dir: &'static Dir<'static>) -> Result<Vec<M<'static>>> {
-    // We want to limit the number of allocations here
-    let mut btreemap = BTreeMap::new();
+    let mut migrations: Vec<Option<M>> = vec![None; dir.dirs().count() + 1];
```

Comment:
After checking, it looks like `dir` already holds the file list, so this does not have extra costs compared to calling `dir.dirs()` just once (currently, the second call is a few lines down, in the for).

Reply:
So all good to keep this as is.